apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/db/DatabaseManagerImpl.java

```diff
@@ -185,23 +215,41 @@ public TransactionalDataSource createAndAddShard(Long shardId, DbVersion dbVersi
     }
 
     @Override
-    public synchronized List<TransactionalDataSource> getFullDataSources() {
+    public synchronized List<TransactionalDataSource> getFullDataSources(Long numberOfShards) {
```

Comment:
There are ambiguous method names.
getFullDataSources - Does it mean 'all' data sources?

Reply:
All FULL data sources, because we also can have HALF data sources (data imported from ZIP archive)